tests/integration/targets/meraki_mx_intrusion_prevention/tasks/tests.yml

```diff
@@ -199,114 +212,121 @@
   #   delegate_to: localhost
   #   register: full_idempotent
 
-  # - debug:
+  # - ansible.builtin.debug:
   #     var: full_idempotent
 
-  # - assert:
+  # - ansible.builtin.assert:
   #     that:
   #       - full_idempotent is not changed
   #       - full_idempotent.data.mode is defined
 
   - name: Query IPS info for network
-    meraki_intrusion_prevention:
-      auth_key: '{{auth_key}}'
+    cisco.meraki.meraki_mx_intrusion_prevention:
+      auth_key: '{{ auth_key }}'
       state: query
-      org_name: '{{test_org_name}}'
-      net_name: '{{test_net_name}} - IPS'
+      org_name: '{{ test_org_name }}'
+      net_name: '{{ test_net_name }} - IPS'
     register: query_net
 
-  - assert:
+  - name: Assert query_net
+    ansible.builtin.assert:
       that:
         - query_net is defined
 
   - name: Test use_default error with included_cidr
-    meraki_intrusion_prevention:
-      auth_key: '{{auth_key}}'
+    cisco.meraki.meraki_mx_intrusion_prevention:
+      auth_key: '{{ auth_key }}'
       state: present
-      org_name: '{{test_org_name}}'
-      net_name: '{{test_net_name}} - IPS'
+      org_name: '{{ test_org_name }}'
+      net_name: '{{ test_net_name }} - IPS'
       mode: prevention
       ids_rulesets: security
       protected_networks:
         use_default: false
         excluded_cidr:
           - 10.0.1.0/24
-    check_mode: yes
+    check_mode: true
     register: included_missing
-    ignore_errors: yes
+    failed_when: false
+    changed_when: false
 
-  - assert:
+  - name: Assert included_missing
+    ansible.builtin.assert:
       that:
         - 'included_missing.msg == "included_cidr is required when use_default is False."'
 
   - name: Test use_default error with included_cidr
-    meraki_intrusion_prevention:
-      auth_key: '{{auth_key}}'
+    cisco.meraki.meraki_mx_intrusion_prevention:
+      auth_key: '{{ auth_key }}'
       state: present
-      org_name: '{{test_org_name}}'
-      net_name: '{{test_net_name}} - IPS'
+      org_name: '{{ test_org_name }}'
+      net_name: '{{ test_net_name }} - IPS'
       mode: prevention
       ids_rulesets: security
       protected_networks:
         use_default: false
         included_cidr:
           - 10.0.1.0/24
-    check_mode: yes
+    check_mode: true
     register: excluded_missing
-    ignore_errors: yes
+    failed_when: false
```

Comment:
I hadn't considered this. What's the reason for adding this here instead of in an assertion? Could be a good approach.

Reply:
I looked at the site (Ansible Docs)[https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_error_handling.html#defining-failure} and I found out at the begging of my journey with warnings :D

Reply:
I'm thinking this through more and I'd like to understand why we're including those two items in the tasks. My concern is that if we're overriding when it fails and when it is changed, we will be masking other problems.

Reply:
As far I thought we can use the `failed_when` to get information when the task should be fail.

Reply:
We can. I think `failed_when` specifies when a task should fail. But that's what the assertions are for. I'd rather keep the assertions responsible for checking whether a test fails or not. Please remove the `failed_when` and `changed_when` instances.